### ISSUE TITLE
ToggleNode should use Node.swapVisibility

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/sun#324]
From https://github.com/phetsims/plinko-probability/issues/104, ToggleNode should manage focus for a11y. Node.swapVisibility was added to replace visibility of two Nodes and manage focus, so ToggleNode can make use of this.

**Comment by jessegreenberg** [comment]
@samreid since you are listed as an author would you mind reviewing?

**Comment by samreid** [comment]
On hold pending discussion in https://github.com/phetsims/scenery/issues/694#issuecomment-340331112

**Comment by jessegreenberg** [comment]
Discussion is complete in https://github.com/phetsims/scenery/issues/694, we changed swapVisibility to look like `node.swapVisibility( otherNode )`. Visibility of `node` and `otherNode` cannot be equal, but otherwise there are no restrictions to which node is visible before calling the function. @samreid removing the 'on-hold' label but feel free to wait until final review of https://github.com/phetsims/scenery/issues/694 is complete if you prefer.

**Comment by samreid** [comment]
ToggleNode will need to override `dispose` to avoid a memory leak because the constructor introduced a new listener to a non-owned Property.

I tested plinko probability with ?accessibility and it appeared that focus transferred properly to the grayed out play button when I pressed all + play.

**Comment by jessegreenberg** [comment]
Thanks @samreid, I added dispose to ToggleNode, and also added ToggleNode to MemoryTestsScreenView to test for ToggleNode leaks. Before adding dispose you can see a climbing JS Heap in between major garbage collections:
![capture1](https://user-images.githubusercontent.com/6396244/32394764-ddec0bc8-c0a3-11e7-97c7-5c90eee0221f.PNG)

And after dispose was added that goes away:
![capture3](https://user-images.githubusercontent.com/6396244/32394790-f8297246-c0a3-11e7-9e0d-9bb8e0d0cd59.PNG)

@samreid can this be closed?

**Comment by samreid** [comment]
Looks great, thanks, closing.
